### ISSUE TITLE
raar-record-handler - Add effective track duration to recording file

### ISSUE THREAD

**Issue by paraenggu** [radiorabe/raar#6]
Currently, `rotter` is configured [to add a fixed duration of 60 minutes to every file name](https://github.com/radiorabe/raar/blob/6206ff03f4c52582239b6113926774e4ff427a6a/config/systemd/rotter%40raar.service.d/override.conf#L3) (`%%FT%%H%%M%%S%%z_060.flac`), regardless of the actual track duration.

`rotter` creates a new recording file at the start of the new hour. During the initial start or in case a recording fails, the recording file won't have a duration of 60 minutes.

To cope with these edge-cases, the following changes should be made:
1. Remove the fixed `_060` duration suffix from the [rotter startup configuration](https://github.com/radiorabe/raar/blob/6206ff03f4c52582239b6113926774e4ff427a6a/config/systemd/rotter%40raar.service.d/override.conf#L3)
2. After a recording has finished, analyze the duration (for example with `ffprobe`) within the `raar-record-handler` script
3. Add the effective duration to the file name before [moving the source file to its temporary and final location](https://github.com/radiorabe/raar/blob/6206ff03f4c52582239b6113926774e4ff427a6a/bin/raar-record-handler.sh#L87-L106)

The duration should be in the format defined by [ISO 8601 durations](https://en.wikipedia.org/wiki/ISO_8601#Durations) or specified as `_MMM` (in minutes). The raar importer recently gained support for parsing a duration with e8268630dea56dbb3b436fa016d3ede594ef7a2a

**Comment by paraenggu** [comment]
@codez
I'm planning to use [`ffprobe`](http://www.ffmpeg.org/ffprobe.html) to get the duration of the length of a recording which will output the duration in seconds with microsecond accuracy.
```bash
ffprobe -i 2019-11-30T190000+0100.flac \
        -show_entries format="duration" \
        -print_format csv="print_section=0" \
        -v quiet 
```
```
278.813333
```
The above example file has a duration of 278.813333 seconds.

The duration will then be added as a suffix to the existing file name in seconds in the [ISO 8601 durations](https://en.wikipedia.org/wiki/ISO_8601#Durations) format:
```
# Variant A) Including microseconds
2019-11-30T190000+0100_PT278.813333S.flac

# Variant B) without microsecond accuracy:
2019-11-30T190000+0100_PT278S.flac
```

As far as I understand e8268630dea56dbb3b436fa016d3ede594ef7a2a this should be possible, but would like to make sure, that you are OK with this format before I'm going to implement it.

Thanks!

**Comment by codez** [comment]
Both are possible for raar. The applied tolerance is 5 seconds, so microseconds are not really used. I guess the readability when listing files on the console is better when the duration is rounded to the closest second.

**Comment by paraenggu** [comment]
Thanks @codez for the prompt feedback!

**Comment by paraenggu** [comment]
Implemented with #7 and successfully tested by the end of December 2019. Closing this issue.
